crates/deltalake-core/src/operations/write.rs

```diff
@@ -314,6 +324,14 @@ pub(crate) async fn write_execution_plan(
     };
 
     let checker = DeltaDataChecker::new(snapshot);
```

Comment:
Couple of reasons I had in mind, which made me choose this approach, happy to reconsider if I missed something more important that would make the mutable implementation more important

1. DeltaDataChecker is using a builder style we used in many places which does not require the builder variable itself to be mutable
2. At the moment I am creating a checker here but if later for some reason it was provided then I would not want to change it in place.

Reply:
Because it uses this builder style it leans itself wel to do this kind of structure, see here: 

https://github.com/delta-io/delta-rs/blob/6d41b37de413f17b28f2b8ece8eb72b0b1d27215/python/src/lib.rs#L102-L106

Reply:
Yes I understand what you mean, it is less verbose but it's just a style in my opinion, but I'd end up with a mutable. Again, happy to change if that's important

Reply:
I'm leaving this as is for now, it's a small change, can be easily done as a small PR if needed